src/server/active_tran_server.cpp

```diff
@@ -48,10 +48,43 @@ active_tran_server::~active_tran_server ()
     }
 }
 
-int active_tran_server::init_page_server_hosts (const char *db_name)
+int active_tran_server::parse_server_host (const std::string &host)
 {
-  assert_is_active_tran_server ();
+  std::string m_ps_hostname;
+  auto col_pos = host.find (":");
+  long port = -1;
+
+  if (col_pos < 1 || col_pos >= host.length () - 1)
+    {
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_HOST_PORT_PARAMETER, 2, prm_get_name (PRM_ID_PAGE_SERVER_HOSTS),
+	      host.c_str ());
+      return ER_HOST_PORT_PARAMETER;
+    }
+
+  try
+    {
+      port = std::stol (host.substr (col_pos + 1));
+    }
+  catch (...)
+    {
+    }
 
+  if (port < 1 || port > USHRT_MAX)
+    {
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_HOST_PORT_PARAMETER, 2, prm_get_name (PRM_ID_PAGE_SERVER_HOSTS),
+	      host.c_str ());
+      return ER_HOST_PORT_PARAMETER;
+    }
+  // host and port seem to be OK
+  m_ps_hostname = host.substr (0, col_pos);
+  er_log_debug (ARG_FILE_LINE, "Page server hosts: %s port: %d\n", m_ps_hostname.c_str (), port);
+
+  cubcomm::node conn{port, m_ps_hostname};
+  m_connection_list.push_back (conn);
+}
```

Comment:
A value must be returned on all paths.
```suggestion
  return NO_ERROR;
}
```